crates/driver/src/domain/competition/auction.rs

```diff
@@ -605,3 +637,66 @@ pub enum Error {
     #[error("blockchain error: {0:?}")]
     Blockchain(#[from] blockchain::Error),
 }
+
+trait OrderingKey: Send + Sync + 'static {
```

Comment:
Would extract all of this into its own `competition::sorting::Strategy` module/component

Reply:
https://github.com/cowprotocol/services/pull/2872